From 8e7167ad0ea919e3af46eebdd1ccdaba3dd93acc Mon Sep 17 00:00:00 2001
From: Mike Gerdts <mike.gerdts@joyent.com>
Date: Fri, 17 Aug 2018 19:15:33 +0000
Subject: [PATCH] OS-7145 lx audit misses bounds check Reviewed by: Patrick
 Mooney <patrick.mooney@joyent.com> Reviewed by: Jerry Jelinek
 <jerry.jelinek@joyent.com> Approved by: Jerry Jelinek
 <jerry.jelinek@joyent.com>

---
 usr/src/uts/common/brand/lx/os/lx_audit.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/usr/src/uts/common/brand/lx/os/lx_audit.c b/usr/src/uts/common/brand/lx/os/lx_audit.c
index de68bcafb1..6e522e6d8d 100644
--- a/usr/src/uts/common/brand/lx/os/lx_audit.c
+++ b/usr/src/uts/common/brand/lx/os/lx_audit.c
@@ -862,6 +862,9 @@ lx_audit_syscall_exit(int sysnum, long ret)
 	if (lxzd->lxzd_audit_enabled == LXAE_DISABLED)
 		return;
 
+	if (sysnum >= LX_NSYSCALLS)
+		return;
+
 	asp = lxzd->lxzd_audit_state;
 	ASSERT(asp != NULL);
 
-- 
2.21.0

